### PR TITLE
Add latest sit script

### PR DESCRIPTION
**Pull request by robin-k-wilson**
**Test Plan for Sit Scripts:**

Setup:

1. Create sittable entity, can either:
     - Import from json: [Download Blue Chair JSON](https://hifi-production.s3.amazonaws.com/robin/sit/sitPrototypes/bluechairlatest.json)
     - Or add to an entity. Use the registration point to be the seat center. Ex - a stool would use a registration point of { x: 0.5, y: 1.0, z: 0.5 }
2. Ensure both the entity script and server script is running on Entity
3. Spawn multiple chairs. Best tested with multiple avatars and users see "Multiple chair/user" test section

**Desktop regular use case**
1. Walk within 3 m from the chair
2. “Click to Sit” overlay appears on chair flat on the seat and in the chair direction
3. Should fade to a barely visible dark grey over time
4. Click on the overlay
5. Pop into chair facing forward (without HMD overlay) and sit
6. Can swivel around in chair with mouse
7. Use any movement key and a “Hold to Stand” overlay will appear in front of view
8. Should pop in front of the chair facing the direction the avatar was facing

**HMD regular use case - differences from Desktop in bold**
1. Walk within 3 m from the chair
2. **“Trigger to Sit”** overlay appears on chair flat on the seat and in the chair direction
3. Should fade to a barely visible dark grey over time
4. Trigger on the overlay
5. **Overlay to Face Forward appears for one second for user to get comfortable in real space**
6. **Timed at the end of the overlay, Your avatar head and camera should be facing forward in the chair with your legs facing forward**
7. Can **snap** and rotate around in chair with **right joystick**
8. **Head can look around normally without popping out of the chair**
9. Use any movement key and a “Hold to Stand” overlay will appear in front of view
10. Should pop in front of the chair facing the direction the avatar was facing

**Overlay tests**
- Sittable overlay appears within 2.8 m of walking to the chair disappears after walking outside 5 m
- Face Forward and Hold to Stand overlay should stay in middle of view
- In HMD mode always see Trigger to Sit, in Desktop mode always see Click to Sit

**Get out of chair tests**
- In HMD, can move away from old position and pop out of chair
- Hold down movement keys including flying
- Disconnects from server then reloads it
- Pops out of chair when:
   - Resize avatar
   - Change Avatar
   - Change domain
   - Teleport out of chair

**Multiple chair/user tests**
- Overlay 3m/5m apply
- While sitting, can switch chairs using the overlay. Old chair become available for another person
- Click/trigger to sit overlay disappears shortly after another HMD user triggers to sit on chair

**Outlier cases to ensure do not happen**
- Avatar never has the sit animation active outside of a chair 
- 2 avatars can not sit in the same chair _- see Known Bug section_
- Should not see sittable overlay when another avatar is sitting in chair
- Chair does not lock up. If avatar disconnects suddenly, the heartbeat interval on the server script should allow chair to be settable again within about 11 seconds.

**Known Bugs - that happen <5% of the time**
- In HMD mode, after sit, avatar sometimes is not rotated forward in the chair
- 2 avatars sit in the same chair if they trigger/click at the exact same time
- Significant lag in a domain can cause an avatar in HMD to pop out of chair without sitting - can correct by trying again

### CODE REVIEW

**Comment by hyperlogic** [comment]
@robin-k-wilson Please run eslint on this script, we want to ensure that this follows our standards.  Here is the list of eslint errors that need to be resolved:

```
e:\code\hifi\examples\robin-sit.js:19:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:20:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:27:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:29:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:31:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:33:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:34:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:35:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:36:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:38:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:40:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:41:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:43:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:44:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:45:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:46:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:47:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:48:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:50:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:51:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:53:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:54:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:56:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:57:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:59:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:60:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:60:9: 'SITTING_SEARCH_RADIUS' is assigned a value but never used. [Error/no-unused-vars]
e:\code\hifi\examples\robin-sit.js:62:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:64:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:65:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:66:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:68:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:77:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:78:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:79:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:80:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:83:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:84:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:85:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:86:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:87:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:88:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:89:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:91:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:92:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:93:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:94:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:94:129: Missing semicolon. [Error/semi]
e:\code\hifi\examples\robin-sit.js:95:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:97:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:98:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:99:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:100:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:102:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:104:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:105:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:107:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:109:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:170:14: Unexpected trailing comma. [Error/comma-dangle]
e:\code\hifi\examples\robin-sit.js:241:124: Unexpected trailing comma. [Error/comma-dangle]
e:\code\hifi\examples\robin-sit.js:242:22: Missing semicolon. [Error/semi]
e:\code\hifi\examples\robin-sit.js:254:126: Unexpected trailing comma. [Error/comma-dangle]
e:\code\hifi\examples\robin-sit.js:257:14: Unexpected trailing comma. [Error/comma-dangle]
e:\code\hifi\examples\robin-sit.js:313:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:343:13: Expected { after 'if' condition. [Error/curly]
e:\code\hifi\examples\robin-sit.js:349:13: Expected { after 'if' condition. [Error/curly]
e:\code\hifi\examples\robin-sit.js:366:61: 'HALF' is not defined. [Error/no-undef]
e:\code\hifi\examples\robin-sit.js:377:1: Line 377 exceeds the maximum line length of 128. [Error/max-len]
e:\code\hifi\examples\robin-sit.js:383:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:385:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:392:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:407:9: Expected { after 'if' condition. [Error/curly]
e:\code\hifi\examples\robin-sit.js:408:9: Expected { after 'if' condition. [Error/curly]
e:\code\hifi\examples\robin-sit.js:409:9: Expected { after 'if' condition. [Error/curly]
e:\code\hifi\examples\robin-sit.js:413:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:418:15: No magic number: 4. [Error/no-magic-numbers]
e:\code\hifi\examples\robin-sit.js:422:15: No magic number: 5. [Error/no-magic-numbers]
e:\code\hifi\examples\robin-sit.js:446:19: No magic number: 6. [Error/no-magic-numbers]
e:\code\hifi\examples\robin-sit.js:470:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:506:6: Unnecessary semicolon. [Error/no-extra-semi]
e:\code\hifi\examples\robin-sit.js:508:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:522:12: No magic number: 50. [Error/no-magic-numbers]
e:\code\hifi\examples\robin-sit.js:560:6: Unnecessary semicolon. [Error/no-extra-semi]
e:\code\hifi\examples\robin-sit.js:563:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:576:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:595:71: Expected '!==' and instead saw '!='. [Error/eqeqeq]
e:\code\hifi\examples\robin-sit.js:677:6: Unnecessary semicolon. [Error/no-extra-semi]
e:\code\hifi\examples\robin-sit.js:679:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:682:5: Expected indentation of 5 spaces but found 4. [Error/indent]
e:\code\hifi\examples\robin-sit.js:693:1: Expected indentation of 1 space but found 0. [Error/indent]

90 problems
```

File: Prototyping/Sit/sit.js [hunk 1]
```diff
@@ -0,0 +1,692 @@
...
+                var distanceFromSeat = Vec3.distance(MyAvatar.position, seatPosition);
+                return distanceFromSeat < OVERLAY_SITTABLE_DISTANCE_SHOW && !utils.checkSeatForAvatar();
+            },
+            // This part is used to fade out the overlay over time
+            lerpTransparency: function () {
+                var startAlpha = OVERLAY_SITTABLE_ALPHA_START;
+                var changeAlpha = 0.01;
+                overlayIntervalTransparency = Script.setInterval(function () {
+                    startAlpha = startAlpha - changeAlpha; // My new alpha
+                    Overlays.editOverlay(overlaySittable, { alpha: startAlpha }); // Edit the existing overlay
```
Comment (hyperlogic) [inline Prototyping/Sit/sit.js]: you should probably check that overlaSittable is not null, before calling editOveraly.

File: Prototyping/Sit/sit.js [hunk 2]
```diff
@@ -0,0 +1,692 @@
...
+        },
+
+        preSit: {
+            preload: function () {
+                var str;
+                for (var i = 0; i < OVERLAY_PRESIT_URL_NUM; i++) {
+                    str = i + 1;
+                    overlayPreSitLoaded[i] = TextureCache.prefetch(OVERLAY_PRESIT_URL_ROOT + str + OVERLAY_PRESIT_URL_POSTFIX);
+                }
+                OVERLAY_PRESIT_URL_TEXT = TextureCache.prefetch(OVERLAY_PRESIT_URL_TEXT);
```
Comment (hyperlogic) [inline Prototyping/Sit/sit.js]: This is bad form, in our coding convention standard a variable with ALL_CAPS is assumed to be a constant.  but here are you are changing it.  I would use a new varaible named overlayPresetTexture or something instead instead of changing the value of OVERLAY_PRESIT_URL_TEXT.

File: Prototyping/Sit/sit.js [hunk 3]
```diff
@@ -0,0 +1,692 @@
...
+
+                    var updateObj = {
+                        rotation: cameraRotation,
+                        position: Vec3.sum(Camera.position, Vec3.multiplyQbyV(Camera.orientation, { x: 0, y: 0.1, z: -1 })),
+                    }
+
+                    var timePassed = now - overlayPreSitStart;
+                    if (timePassed >= OVERLAY_PRESIT_FRAME_DURATION) {
+                        overlayPreSitStart = now;
+                        index = index + 1;
```
Comment (hyperlogic) [inline Prototyping/Sit/sit.js]: Please give this variable a more descriptive name, I'm concerned that someone might inadvertently change this, because we use index as a variable in several other places in this file.  How about preSitLoadIndex?

File: Prototyping/Sit/sit.js [hunk 4]
```diff
@@ -0,0 +1,692 @@
...
+                    });
+                }
+            }
+        },
+
+        showSittable: function () { // Show overlays when I'm close to the seat
+
+            if (isInEditMode()) {
+                return;
+            }
```
Comment (hyperlogic) [inline Prototyping/Sit/sit.js]: Is returning what we want to do here?  I thought maybe we should hide the overlay instead?

File: Prototyping/Sit/sit.js [hunk 5]
```diff
@@ -0,0 +1,692 @@
...
+        }
+    };
+
+    var utils = {
+
+        // Is the seat used
+        checkSeatForAvatar: function () {
+            return deadlockGate;
+
+            // var isOccupied = false;
```
Comment (hyperlogic) [inline Prototyping/Sit/sit.js]: Please remove this dead code.

File: Prototyping/Sit/sit.js [hunk 6]
```diff
@@ -0,0 +1,692 @@
...
+            return distanceFromSeat < SITTABLE_DISTANCE_MAX && !this.checkSeatForAvatar();
+        },
+
+        calculatePinHipPosition: function () {
+            if (!chairProperties) this.setChairProperties();
+
+            // get chairProperties to calculate middle of chair
+            seatCenterPosition = chairProperties.position;
+
+            var yOffset = chairProperties.dimensions.y / 2 * chairOffsetRatio;
```
Comment (hyperlogic) [inline Prototyping/Sit/sit.js]: This chairOffsetRatio maybe should be multiplied by some factor depending on the MyAvatar.getHeight().  so that tall avatars will have a large offset and short avatar would have a smaller offset.  For example:
```
var DEFAULT_AVATAR_HEIGHT = 1.755;
var chairOffsetRatio = 0.2 * (MyAvatar.getHeight() / DEFAULT_AVATAR_HEIGHT);
```

File: Prototyping/Sit/sit.js [hunk 7]
```diff
@@ -0,0 +1,692 @@
...
+        if (overlayStandUp) overlays.standUp.remove();
+
+    };
+
+    function sitAndPinAvatar() {
+
+        Settings.setValue(SETTING_KEY, entityID);
+        sitDownSettlePeriod = Date.now() + IK_SETTLE_TIME;
+
+        print(4);
```
Comment (hyperlogic) [inline Prototyping/Sit/sit.js]: please remove these debug statements.  Or add a global variable SITTING_DEBUG and do something like:
```
if (SITTING_DEBUG) {
    print("got here 4");
}
```

File: Prototyping/Sit/sit.js [hunk 8]
```diff
@@ -0,0 +1,692 @@
...
+
+        // MyAvatar.resetSensorsAndBody();
```
Comment (hyperlogic) [inline Prototyping/Sit/sit.js]: remove

File: Prototyping/Sit/sit.js [hunk 9]
```diff
@@ -0,0 +1,692 @@
...
+
+        print(5);
```
Comment (hyperlogic) [inline Prototyping/Sit/sit.js]: remove

File: Prototyping/Sit/sit.js [hunk 10]
```diff
@@ -0,0 +1,692 @@
...
+                ANIMATION_FIRST_FRAME, ANIMATION_LAST_FRAME);
+        }
+        for (var i in OVERRIDDEN_DRIVE_KEYS) {
+            MyAvatar.disableDriveKey(OVERRIDDEN_DRIVE_KEYS[i]);
+        }
+
+        MyAvatar.centerBody();
+        var index = MyAvatar.getJointIndex("Hips");
+
+        // var hipLocalRotation = MyAvatar.getJointRotation(index);
```
Comment (hyperlogic) [inline Prototyping/Sit/sit.js]: remove this dead code.

File: Prototyping/Sit/sit.js [hunk 11]
```diff
@@ -0,0 +1,692 @@
...
+        // var hipWorldRotation = Quat.multiply(MyAvatar.orientation, hipLocalRotation);
+        // var yawHipWorldRotation = Quat.cancelOutRollAndPitch(hipWorldRotation);
+
+        Script.setTimeout(function () {
+
+            print(6, JSON.stringify(chairProperties.rotation));
```
Comment (hyperlogic) [inline Prototyping/Sit/sit.js]: remove

File: Prototyping/Sit/sit.js [hunk 12]
```diff
@@ -0,0 +1,692 @@
...
+            MyAvatar.pinJoint(index, seatCenterPosition, chairProperties.rotation);
+            canStand = true;
+
+            Script.update.connect(update);
+            MyAvatar.scaleChanged.connect(standUp);
+            location.hostChanged.connect(standUp);
+
+        }, SIT_DELAY);
+
+        animStateHandlerID = MyAvatar.addAnimationStateHandler(function (avatarProperties) {
```
Comment (hyperlogic) [inline Prototyping/Sit/sit.js]: Why is this here? We should not be directly changing how the IK works.  please remove.

File: Prototyping/Sit/sit.js [hunk 13]
```diff
@@ -0,0 +1,692 @@
...
+            return { headType: 0 };
+        }, ["headType"]);
+
+    }
+
+    function sitDown() {
+        print("SIT DOWN");
```
Comment (hyperlogic) [inline Prototyping/Sit/sit.js]: remove

File: Prototyping/Sit/sit.js [hunk 14]
```diff
@@ -0,0 +1,692 @@
...
+
+        // if (utils.checkSeatForAvatar()) {
```
Comment (hyperlogic) [inline Prototyping/Sit/sit.js]: remove

File: Prototyping/Sit/sit.js [hunk 15]
```diff
@@ -0,0 +1,692 @@
...
+
+        var avatarDistance = Vec3.distance(MyAvatar.position, seatCenterPosition);
+
+
+        Script.setTimeout(function () {
+            // Move avatar in front of the chair to avoid getting stuck in collision hulls
+            if (avatarDistance < STANDUP_DISTANCE) {
+                var index = MyAvatar.getJointIndex("Hips");
+                MyAvatar.clearPinOnJoint(index);
+                var offset = { x: 0, y: 1.0, z: -1 / 2 - chairProperties.dimensions.z * chairProperties.registrationPoint.z };
```
Comment (hyperlogic) [inline Prototyping/Sit/sit.js]: the -1/2 should be replaced by a constant. like CHAIR_DISMOUNT_OFFSET or something.

File: Prototyping/Sit/sit.js [hunk 16]
```diff
@@ -0,0 +1,692 @@
...
+                this.sittable.remove();
+            }
+        }
+    };
+
+    var utils = {
+
+        // Is the seat used
+        checkSeatForAvatar: function () {
+            return deadlockGate;
```
Comment (hyperlogic) [inline Prototyping/Sit/sit.js]: I'm confused is this used to check if the MyAvatar is in the chair? or if ANY avatar is sitting in the chair.

File: Prototyping/Sit/sit.js [hunk 17]
```diff
@@ -0,0 +1,692 @@
...
+        //     return;
+        // }
+
+        if (overlaySittable !== null) {
+            overlays.sittable.remove();
+        }
+
+        utils.setHeadToHipsDistance();
+
+        lockChairOnStandUp = Entities.getEntityProperties(entityID, 'locked').locked;
```
Comment (hyperlogic) [inline Prototyping/Sit/sit.js]: Why do we need to lock/unlock the chair, it doesn't look like we are modifying it at all.

File: Prototyping/Sit/sit.js [hunk 18]
```diff
@@ -0,0 +1,692 @@
...
+        }
+        driveKeyPressedStart = null;
+        overlays.standUp.remove();
+
+        Script.update.disconnect(update);
+        MyAvatar.scaleChanged.disconnect(standUp);
+        location.hostChanged.disconnect(standUp);
+
+        canStand = false;
+        MyAvatar.removeAnimationStateHandler(animStateHandlerID);
```
Comment (hyperlogic) [inline Prototyping/Sit/sit.js]: remove

File: Prototyping/Sit/sit.js [hunk 19]
```diff
@@ -0,0 +1,692 @@
...
+
+        }
+        sittingDown = false;
+        deadlockGate = false;
+        sitDownSettlePeriod = null;
+    };
+
+    // User can also click on overlay to sit down
+    this.mouseReleaseOnOverlay = function (overlayID, pointerEvent) {
+        print("Mouse release");
```
Comment (hyperlogic) [inline Prototyping/Sit/sit.js]: remove print

File: Prototyping/Sit/sit.js [hunk 20]
```diff
@@ -0,0 +1,692 @@
...
+        if (overlayID === overlaySittable && pointerEvent.isLeftButton) {
+            print("CAN SIT", deadlockGate, Settings.getValue(SETTING_KEY));
```
Comment (hyperlogic) [inline Prototyping/Sit/sit.js]: remove print

File: Prototyping/Sit/sit.js [hunk 21]
```diff
@@ -0,0 +1,692 @@
...
+            var avatarDistance = Vec3.distance(MyAvatar.position, seatCenterPosition);
+            if (avatarDistance > DISTANCE_FROM_CHAIR_TO_STAND_AT && sitDownSettlePeriod && now > sitDownSettlePeriod) {
+                hasAvatarMovedTooFar = true;
+            }
+
+            // AVATAR SPINE ERROR
+            if (HMD.active) {
+
+                var headWorldPosition;
+                if (MyAvatar.getJointIndex("Head") !== -1) {
```
Comment (hyperlogic) [inline Prototyping/Sit/sit.js]: I don't think we need to check if the avatar has a head here.  If we are in HMD mode we are guaranteed that getPoseValue(Head) will return something.

File: Prototyping/Sit/sit.js [hunk 22]
```diff
@@ -0,0 +1,692 @@
...
+
+                    // if head is more than certain distance from hips or less than certain distance, stand up
+                    var headPoseValue = Controller.getPoseValue(Controller.Standard.Head);
+                    headWorldPosition = Vec3.sum(Vec3.multiplyQbyV(MyAvatar.orientation, headPoseValue.translation), 
+                        MyAvatar.position);
+                } else {
+
+                    // Avatar has no head joint
+                    print("I haz no head!!!");
```
Comment (hyperlogic) [inline Prototyping/Sit/sit.js]: remove print

File: Prototyping/Sit/sit.js [hunk 23]
```diff
@@ -0,0 +1,692 @@
...
+                    var tempPosition = MyAvatar.position;
+                    tempPosition.y += headToHipsDistance;
+                    headWorldPosition = tempPosition; // this is just sitting up straight, not calculating head position
+                }
+
+                var headDeviation = Vec3.distance(headWorldPosition, seatCenterPosition);
+                // print("headDeviation is ", headDeviation);
```
Comment (hyperlogic) [inline Prototyping/Sit/sit.js]: remove dead prints

File: Prototyping/Sit/sit.js [hunk 24]
```diff
@@ -0,0 +1,692 @@
...
+                    if (now - deviationTimeStart > deviationTimeToStand) {
+                        hasAvatarSpineError = true;
+                        deviationTimeStart = null;
+                    }
+                }
+            }
+
+            if (hasAvatarSpineError || hasHeldDriveKey || hasAvatarMovedTooFar) {
+                standUp();
+                print("ROBIN ", hasAvatarSpineError, hasHeldDriveKey, hasAvatarMovedTooFar);
```
Comment (hyperlogic) [inline Prototyping/Sit/sit.js]: remove print

File: Prototyping/Sit/sit.js [hunk 25]
```diff
@@ -0,0 +1,692 @@
...
+
+                var headDeviation = Vec3.distance(headWorldPosition, seatCenterPosition);
+                // print("headDeviation is ", headDeviation);
+                // print("headToHipsDistance: ", headToHipsDistance);
+                var deviationRatio = headDeviation/headToHipsDistance;
+
+                if ((deviationRatio > ONE_HUNDRED_AND_TEN_PERCENT || deviationRatio < EIGHTY_PERCENT) && 
+                    sitDownSettlePeriod && now > sitDownSettlePeriod) {
+
+                    // print("large deviation");
```
Comment (hyperlogic) [inline Prototyping/Sit/sit.js]: remove print

File: Prototyping/Sit/sit.js (hunk 6 above)
Comment (robin-k-wilson) [reply Prototyping/Sit/sit.js]: Flagging this comment to come back to.

File: Prototyping/Sit/sit.js (hunk 12 above)
Comment (robin-k-wilson) [reply Prototyping/Sit/sit.js]: Flagging for question on this function.

**Comment by robin-k-wilson** [comment]
- Fixed linter errors and removed prints
- Changed variable names as mentioned by hyperlogic
- Added sitServer.js script to ensure only one Avatar can sit at a time
- Overlay updates depending on HMD vs Desktop mode

File: Prototyping/Sit/sit.js [hunk 26]
```diff
@@ -0,0 +1,691 @@
...
+
+            sittingDown = true;
+
+            MyAvatar.pinJoint(index, seatCenterPosition, chairProperties.rotation);
+            canStand = true;
+
+            Script.update.connect(update);
+            MyAvatar.scaleChanged.connect(standUp);
+            location.hostChanged.connect(standUp);
+            Script.scriptEnding.connect(standUp);
```
Comment (hyperlogic) [inline Prototyping/Sit/sit.js]: I think we also want to standUp if the MyAvatar.onLoadComplete signal is fired

File: Prototyping/Sit/sit.js [hunk 27]
```diff
@@ -0,0 +1,691 @@
...
+
+            sittingDown = false;
+            sitDownSettlePeriod = null;
+        
+        }
+
+        Script.update.disconnect(update);
+        MyAvatar.scaleChanged.disconnect(standUp);
+        location.hostChanged.disconnect(standUp);
+        Script.scriptEnding.disconnect(standUp);
```
Comment (hyperlogic) [inline Prototyping/Sit/sit.js]: And here disconnect from MyAvatar.onLoadComplete

File: Prototyping/Sit/sit.js [hunk 28]
```diff
@@ -0,0 +1,691 @@
...
+
+                    var deviationTimeToStand = 500;
+                    if (now - deviationTimeStart > deviationTimeToStand) {
+                        hasAvatarSpineError = true;
+                        deviationTimeStart = null;
+                    }
+                }
+            }
+
+            if (SITTING_DEBUG){
```
Comment (hyperlogic) [inline Prototyping/Sit/sit.js]: formatting

**Comment by AndySeattle** [comment]
@robin-k-wilson I tried to download the JSON in the test plan above, and could not, i get a "failed - Forbidden" meaning access denied to the file. is it possible because this file was added before i was part of the group?

**Comment by AndySeattle** [comment]
Howdy, @robin-k-wilson still seeing the same "failed"forbidden" error when I try to download the JSON file. This is blocking me from getting started, so going to mark this as QA failed to bubble up a bit :) 

image

**Comment by AndySeattle** [comment]
Have a prototype of this running in my office space now, testing it out.

**Comment by misslivirose** [comment]
@robin-k-wilson @AndySeattle are these the versions we're using actively right now? Can we get this PR finished up and merged?

**Comment by AndySeattle** [comment]
We just reviewed and this and looks good, QA approved.

**Comment by misslivirose** [comment]
Thank you!
